src/dispatchers/browserContextDispatcher.ts

```diff
@@ -63,6 +64,27 @@ export class BrowserContextDispatcher extends Dispatcher<BrowserContext, channel
         this._dispatchEvent('serviceWorker', { worker: new WorkerDispatcher(this._scope, serviceWorker)});
       context.on(CRBrowserContext.CREvents.ServiceWorker, serviceWorker => this._dispatchEvent('serviceWorker', { worker: new WorkerDispatcher(this._scope, serviceWorker) }));
     }
+    context.on(BrowserContext.Events.Request, (request: Request) =>  {
+      return this._dispatchEvent('request', {
+        request: RequestDispatcher.from(this._scope, request),
+        page: request.frame()._page.initializedOrNull()
```

Comment:
@dgozman I tried with `PageDispatcher.fromNullable(scope, request.frame()._page.initializedOrNull())`, dont think its worthing as expected. 

```
   Error:
        at Object.assert (/Users/vigneshh/opensource/playwright/src/utils/utils.ts:67:11)
        at new Dispatcher (/Users/vigneshh/opensource/playwright/src/dispatchers/dispatcher.ts:67:5)
        at new PageDispatcher (/Users/vigneshh/opensource/playwright/src/dispatchers/pageDispatcher.ts:51:5)
        at CRBrowserContext.<anonymous> (/Users/vigneshh/opensource/playwright/src/dispatchers/browserContextDispatcher.ts:53:88)
        at CRBrowserContext.emit (events.js:315:20)
        at Page.reportAsNew (/Users/vigneshh/opensource/playwright/src/server/page.ts:198:26)
        at Page.initializedOrNull (/Users/vigneshh/opensource/playwright/src/server/page.ts:207:10)
        at CRBrowserContext.<anonymous> (/Users/vigneshh/opensource/playwright/src/dispatchers/browserContextDispatcher.ts:70:72)
        at CRBrowserContext.emit (events.js:315:20)
        at FrameManager.requestStarted (/Users/vigneshh/opensource/playwright/src/server/frames.ts:266:32)
```

Reply:
`PageDispatcher.fromNullable(scope, request.frame()._page.initializedOrNull())` looks very good, just use `this._scope` instead of `scope`. The error is due to the `reportAsNew` vs `initializedOrNull` - see my comment there.

Reply:
Still the same error, Is there a race condition here? Trying to dispatch to the page but connection isn't established still? https://github.com/microsoft/playwright/blob/dddfbaaeb257ca80e07e4aed3bb6d3f9b90923fd/src/dispatchers/dispatcher.ts#L67

Reply:
I believe the problem is that we call `reportAsNew` twice. Just drop `this.reportAsNew()` from the `initializedOrUndefined` method - that should fix it.

Reply:
Ahh nice, totally missed that. Thanks that fixes the issue.